src/core.js

```diff
@@ -1300,12 +1300,12 @@ $.extend( $.validator, {
 			this.settings.messages[ element.name ].remote = previous.message;
 
 			param = typeof param === "string" && { url: param } || param;
-
-			if ( previous.old === value ) {
+			optionDataString = JSON.stringify( $.param( $.extend( { data: value }, param.data ) ) );
```

Comment:
it converts a value into a json string, even objects and arrays:
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify